VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -73,23 +73,54 @@ IOReturn VoodooI2CHIDDevice::parseHIDDescriptor() {
     }
 
     OSDictionary* property_array = OSDictionary::withCapacity(1);
-    property_array->setObject("HIDDescLength", OSNumber::withNumber(hid_descriptor.wHIDDescLength, 32));
-    property_array->setObject("BCDVersion", OSNumber::withNumber(hid_descriptor.bcdVersion, 32));
-    property_array->setObject("ReportDescLength", OSNumber::withNumber(hid_descriptor.wReportDescLength, 32));
-    property_array->setObject("ReportDescRegister", OSNumber::withNumber(hid_descriptor.wReportDescRegister, 32));
-    property_array->setObject("MaxInputLength", OSNumber::withNumber(hid_descriptor.wMaxInputLength, 32));
-    property_array->setObject("InputRegister", OSNumber::withNumber(hid_descriptor.wInputRegister, 32));
-    property_array->setObject("MaxOutputLength", OSNumber::withNumber(hid_descriptor.wMaxOutputLength, 32));
-    property_array->setObject("OutputRegister", OSNumber::withNumber(hid_descriptor.wOutputRegister, 32));
-    property_array->setObject("CommandRegister", OSNumber::withNumber(hid_descriptor.wCommandRegister, 32));
-    property_array->setObject("DataRegister", OSNumber::withNumber(hid_descriptor.wDataRegister, 32));
-    property_array->setObject("VendorID", OSNumber::withNumber(hid_descriptor.wVendorID, 32));
-    property_array->setObject("ProductID", OSNumber::withNumber(hid_descriptor.wProductID, 32));
-    property_array->setObject("VersionID", OSNumber::withNumber(hid_descriptor.wVersionID, 32));
+    if (!property_array)
+        return kIOReturnNoMemory;
+
+    OSNumber* hid_desc_length      = OSNumber::withNumber(hid_descriptor.wHIDDescLength,      32);
```

Comment:
This can we be cleaned up with a small function that will take the number, property array, then create the OSNumber, set the property, then finally release the OSNumber.